drivers/serial/uart_rpi_pico.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021, Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/uart.h>
+#include <drivers/pinctrl.h>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+#include <hardware/uart.h>
+#include <hardware/gpio.h>
+#pragma GCC diagnostic pop
+
+struct uart_rpi_config {
+	uart_inst_t *const uart_dev;
+	uint32_t baudrate;
+	const struct pinctrl_dev_config *pcfg;
+};
+
+static int uart_rpi_poll_in(const struct device *dev, unsigned char *c)
+{
+	const struct uart_rpi_config *config = dev->config;
+
+	if (!uart_is_readable(config->uart_dev)) {
+		return -1;
+	}
+
+	*c = (unsigned char)uart_get_hw(config->uart_dev)->dr;
```

Comment:
why not `uart_getc`?

Reply:
Because it calls `uart_read_blocking`, which is blocking